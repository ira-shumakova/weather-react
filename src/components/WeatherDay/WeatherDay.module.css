.weatherDay {
  backdrop-filter: blur(20px);
  height: 27vh;
  background-color: #44ba8f70;
  border: 2px solid #06201396;
  border-radius: 20px;
}

.weatherIcon {
  height: 9rem;
  padding-left: 1rem;
}

.weatherBox {
  padding-top: 0.5rem;
}

.weatherDescription {
  margin: 0px;
  padding-left: 1rem;
}

.firstGrid {
  text-align: center;
}

.lastGrid {
  padding-left: 0.5rem;
}

.firstGrid h1 {
  padding-bottom: 0px;
  margin-bottom: 0px;
  font-size: 1.75rem;
}

.firstGrid h2 {
  padding-top: 0px;
  margin-top: 0px;
}

.paddingTop {
  padding-top: 1.5rem;
}

.paddingTopGrid {
  padding-top: 2rem;
  font-size: 2.5rem;
}

.paddingTopGrid {
  padding-top: 1rem;
}

.circProgress {
  margin: 10vh 42vh;
}