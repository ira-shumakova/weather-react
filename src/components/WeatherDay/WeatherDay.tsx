import { IMainWeather } from "../../dto/types";
import { Grid } from "@mui/material";
import moment from "moment";
import classes from './WeatherDay.module.css';
import iconLink from "../../utils/getIcons";
import {CircularProgress} from "@mui/material";

type Props = {
    weather: IMainWeather,
    city: string,
    isLoading: boolean,
}

const WeatherDay = ({ weather, city, isLoading }: Props) => {
    const convertTime = (time: number) => {
        return moment.unix(time).local().format("HH:mm")
    }
    const convertDate = (time: number) => {
      return moment.unix(time).local().format("HH:mm, Do MMMM YYYY")
    }

    return (
        <div className={classes.weatherDay}>
          {!isLoading &&
            <Grid 
                container spacing={2}
                justifyContent="space-around"
                alignItems="center"
                className={classes.weatherBox}
            >
                <Grid item xs={3} className={classes.firstGrid}>
                    <h1 className={classes.paddingTopGrid}>{city}</h1>
                    <h2>{Math.round(weather.temp)}&deg;C</h2>
                    <p>Updated at {convertDate(weather.dt)}</p>
                </Grid>
                <Grid item xs={3}>
                    <img 
                        src={iconLink(weather.weather[0].icon)} 
                        alt={weather.weather[0].description}  className={classes.weatherIcon} 
                    />
                    <h3 className={classes.weatherDescription}>
                        {weather.weather[0].description}
                    </h3>
                </Grid>
                <Grid item xs={5} className={classes.lastGrid} >
                    <Grid container>
                        <Grid item xs={6} className={classes.paddingTop}>
                          Feels like: {Math.round(weather.feels_like)}&deg;C
                        </Grid>
                        <Grid item xs={6} className={classes.paddingTop}>
                          Wind: {weather.wind_speed} M/S
                        </Grid>
                        <Grid item xs={6} className={classes.paddingTop}>
                          Pressure: {weather.pressure} hPa
                        </Grid>
                        <Grid item xs={6} className={classes.paddingTop}>
                          Humidity: {weather.humidity}%
                        </Grid>
                        <Grid item xs={6} className={classes.paddingTop}>
                          Sunrise: {convertTime(weather.sunrise)}
                        </Grid>
                        <Grid item xs={6} className={classes.paddingTop}>
                          Sunset: {convertTime(weather.sunset)}
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
          }
          {isLoading &&
            <CircularProgress className={classes.circProgress} color="success"/>
          }
        </div>
    )
}

export default WeatherDay;