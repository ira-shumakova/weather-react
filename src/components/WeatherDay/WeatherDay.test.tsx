import { render } from "@testing-library/react";
import WeatherDay from "./WeatherDay";

describe("WeatherDay", () => {
  test("matches snapshot", () => {
    const weather = {
      temp: 14,
      feels_like: 13,
      humidity: 55,
      pressure: 1017,
      sunrise: 1681960920,
      sunset: 1682011401,
      wind_speed: 6.68,
      dt: 1681984800,
      weather: [
        {
          description: "broken clouds",
          icon: "04d",
          main: "Clouds"
        },
      ],
    };
    const city = "Lviv";
    const isLoading = false;
    const {container} = render(<WeatherDay weather={weather} city={city} isLoading={isLoading}/>)
    expect(container).toMatchSnapshot();
  });
});