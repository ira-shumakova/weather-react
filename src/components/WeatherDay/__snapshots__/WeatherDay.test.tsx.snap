// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDay matches snapshot 1`] = `
<div>
  <div
    class="weatherDay"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 weatherBox css-1hx6p0u-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 firstGrid css-1equabv-MuiGrid-root"
      >
        <h1
          class="paddingTopGrid"
        >
          Lviv
        </h1>
        <h2>
          14
          °C
        </h2>
        <p>
          Updated at 
          13:00, 20th April 2023
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <img
          alt="broken clouds"
          class="weatherIcon"
          src="04d.svg"
        />
        <h3
          class="weatherDescription"
        >
          broken clouds
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 lastGrid css-k3f0g4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 paddingTop css-1osj8n2-MuiGrid-root"
          >
            Feels like: 
            13
            °C
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 paddingTop css-1osj8n2-MuiGrid-root"
          >
            Wind: 
            6.68
             M/S
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 paddingTop css-1osj8n2-MuiGrid-root"
          >
            Pressure: 
            1017
             hPa
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 paddingTop css-1osj8n2-MuiGrid-root"
          >
            Humidity: 
            55
            %
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 paddingTop css-1osj8n2-MuiGrid-root"
          >
            Sunrise: 
            06:22
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 paddingTop css-1osj8n2-MuiGrid-root"
          >
            Sunset: 
            20:23
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
